src/js/core/factories/Grid.js

```diff
@@ -1876,12 +1876,12 @@ angular.module('ui.grid')
         p = 0;
 
     self.columns.forEach(function (col) {
-      if (col.sort && col.sort.priority && col.sort.priority > p) {
-        p = col.sort.priority;
+      if (col.sort && col.sort.priority !== 'undefined' && col.sort.priority >= p) {
```

Comment:
I think here you should compare `col.sort.priority` to `undefined` and not `'undefined'` or `typeof col.sort.priority === 'undefined'`